include/mbgl/map/map_observer.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace mbgl {
+
+class MapObserver {
+public:
+    static MapObserver& nullObserver() {
+        static MapObserver mapObserver;
+        return mapObserver;
+    }
+
+    virtual void onRegionWillChange() {}
```

Comment:
Renaming the public API for iOS and macOS would be a backwards-incompatible change that would require a version bump. Fortunately, this PR doesn’t touch the public API (MGLMapView.h) at all, only the internal implementation of MGLMapView (MGLMapView.mm), so no changelog entry is required.